src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+            {
+                AnalyzeParentInvocationOperation(context, propertyReferenceOperation, (IInvocationOperation)argumentOperation.Parent,
+                    isInstance: false, isAsync: false, methodName: null);
+            }
+        }
+
+        private static void AnalyzeParentBinaryOperation(
+            OperationAnalysisContext context, IOperation operation, IBinaryOperation parent, bool isAsync, string? methodName)
+        {
+            bool useRightSideExpression = default;
+            if (!IsLeftCountComparison(parent, out bool shouldNegateIsEmpty))
+            {
+                if (!IsRightCountComparison(parent, out shouldNegateIsEmpty))
+                {
+                    return;
+                }
+
+                useRightSideExpression = true;
+            }
+
+            if (isAsync)
+            {
+                bool shouldNegateAny = !shouldNegateIsEmpty;
+                ReportCA1828(context, shouldNegateAny, useRightSideExpression ? OperationBinaryRight : OperationBinaryLeft, methodName!, parent);
+            }
+            else
+            {
+                ITypeSymbol type = operation.GetInstanceType();
+
+                if (TypeContainsVsibileProperty(context, type, IsEmpty, SpecialType.System_Boolean))
+                {
+                    ReportCA1836(context, useRightSideExpression, shouldNegateIsEmpty, parent);
+                }
+                else
+                {
+                    bool shouldNegateAny = !shouldNegateIsEmpty;
+                    ReportCA1827(context, shouldNegateAny, useRightSideExpression ? OperationBinaryRight : OperationBinaryLeft, methodName!, parent);
+                }
+            }
+        }
+
+        private static void AnalyzeParentInvocationOperation(
+            OperationAnalysisContext context, IOperation operation, IInvocationOperation parent, bool isInstance, bool isAsync, string? methodName)
+        {
+            if (!IsIntEqualsMethod(parent))
+            {
+                return;
+            }
+
+            if (isInstance)
+            {
+                if (!TryGetZeroOrOneConstant(parent.Arguments[0].Value, out var constant) || constant != 0)
+                {
+                    return;
+                }
+            }
+            else if (!TryGetZeroOrOneConstant(parent.Instance, out var constant) || constant != 0)
+            {
+                return;
+            }
+
+            if (isAsync)
+            {
+                ReportCA1828(context, shouldNegate: true, isInstance ? OperationEqualsInstance : OperationEqualsArgument, methodName!, parent);
+            }
+            else
+            {
+                ITypeSymbol type = operation.GetInstanceType();
+
+                if (TypeContainsVsibileProperty(context, type, IsEmpty, SpecialType.System_Boolean))
+                {
+                    ReportCA1836(context, !isInstance, shouldNegate: false, parent);
+                }
+                else
```

Comment:
Avoid reporting `CA1827` if user already has an efficient property access.
```suggestion
            else if (methodName != null)
```